src/libstd/net/ip.rs

```diff
@@ -837,8 +838,15 @@ impl From<Ipv6Addr> for IpAddr {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl fmt::Display for Ipv4Addr {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        const IPV4_BUF_LEN: usize = 16; // Long enough for the longest possible IPv4 address
+        let mut buf = [0u8; IPV4_BUF_LEN];
+        let mut buf_slice = &mut buf[..];
         let octets = self.octets();
-        write!(fmt, "{}.{}.{}.{}", octets[0], octets[1], octets[2], octets[3])
+        write!(buf_slice, "{}.{}.{}.{}", octets[0], octets[1], octets[2], octets[3]).unwrap();
```

Comment:
```suggestion
        write!(buf_slice, "{}.{}.{}.{}", octets[0], octets[1], octets[2], octets[3]).map_err(|_| fmt::Err)?;
```

Reply:
Does this apply to all the unwraps on `write!`s?

Reply:
An unwrap seems like a cleaner approach for something that we assert can never happen.

Reply:
Yes, I forgot that this write will never fail.

Reply:
This assertion should probably be documented somewhere in the code so that others don't fall into the same trap. I also first looked at the `.unwrap()`s, thinking we should do something about it, but seeing that the compiler should optimize out all the code paths changed my mind. A simple comment like "we can unwrap here because we can rule out any `IO::Error`" should suffice.

Reply:
What about this?
```suggestion
        let _ = write!(buf_slice, "{}.{}.{}.{}", octets[0], octets[1], octets[2], octets[3]);
```

Reply:
@lzutao I don't think that that is a good decision. `unwrap` gives the safety here, that there won't be a problem ever (if you for example change the `write!`-backend in the future which could fail under certain circumstances).
I think `unwrap` is totally fine here.

Reply:
I didn't suggest changing the `unwrap()`s, but adding a comment like:
```
//  `unwrap()` is ok as the error cases should be unreachable`
```
so that others reading the source don't have to wonder about it.

Reply:
@llogiq I have added comments to that effect at the start of both display functions